.github/workflows/build-push-container-on-merge.yml

```diff
@@ -0,0 +1,59 @@
+name: Build and Push Container Image On PR Merge
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions:
+  pull-requests: write
+  contents: write
+  issues: write
+
+jobs:
+  if_merged:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Set env TAG
+        run: |
+          if [ ${{ github.event.pull_request.base.ref }} == "main" ]; then
+              echo "TAG=latest" >> $GITHUB_ENV
+          else
+              echo "TAG=${{ github.event.pull_request.base.ref }}" >> "$GITHUB_ENV"
+          fi
+      - name: Build Image
+        id: build-image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ secrets.IMAGE_NAME }}
+          tags: ${{ env.TAG }}
+          containerfiles: |
+            ./Dockerfile
+
+      - name: Push To Image Registry
+        id: push-to-registry
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          tags: ${{ steps.build-image.outputs.tags }}
+          registry: ${{ secrets.IMAGE_REGISTRY }}/${{ secrets.IMAGE_NAMESPACE }}
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+
+      - name: Set env PR_COMMENT
+        run: |
+          echo "PR_COMMENT=Successfully built and pushed tag: ${{ env.TAG }}" >> $GITHUB_ENV
```

Comment:
what if it failed?

Reply:
1. so the same goes for a successful one, why add only for one flow?
2. how does the user know if `build` failed or `push` failed; should add clear comments on each (so there will be no need to check the logs)

Reply:
Any step that fails, would cause the workflow fail and exit as we are not using continue-on-error. If the workflow fails, we already know it at which step it failed, no?

Reply:
Updated the workflow to add both build and push step status as a comment to the pr.